src/main/java/org/example/DataBaseConnection.java

```diff
@@ -52,8 +52,5 @@ private static DataSource initDataSource(){
         return dataSource;
     }
 
-    public static void main(String[] args) throws SQLException {
```

Comment:
Your method **getConnection** is implemented weirdly. The data source is initialised every time you need a connection, but this is not the desired behaviour. You should init data source only once and get a connection once you need one.